attributes-next.md

```diff
@@ -0,0 +1,252 @@
+  s["passwd"] = "sekret1"
+end
+store("wordpress", "passwd")
+```
+
+## Note on why the New Syntax has split setter/getters
+
+It isn't possible to have a fused setter/getter and to use method chaining.  This API does not work with ruby:
+
+```
+default("foo", "bar", "baz").to("quux")
+default("foo", "bar", "baz")
+ => "quux"
+```
+
+The problem is going to be values like nil and false which would need to be wrapped to implement `#to` methods in
+a decorator to allow writing which would be a truthy return value for the getter and not falsey.
+
+## Removal of Method Missing Syntax
+
+A deprecation warning for method missing syntax access to the node attribute has already been added
+to Chef 12.  In Chef 13, method missing syntax will result in NoMethodError.
+
+## Deprecation of Array Merging
+
+Before the node.save the chef-client will walk the node object and warn on any array objects that are
+being merged with array merging.  Users who are actually using array merging will be encourage to replace
+array merging with hash keys instead.
+
+## Deprecation of Deep Merging with Normal Attributes
+
+Before the node.save the chef-client will walk the node object and warn on any normal attribute which
+also have default, override or automatic attributes set and are being merged.
+
+## Deprecation of Deep Merging with Automatic Attributes
+
+Before the node.save the chef-client will walk the node object and warn on any automatic attribute which
+also have default, normal or override attributes set and are being merged.
+
+## Deprecation of role_default overriding env_default
+
+Before the node.save the chef-client will walk the node object and warn on any env_default attribute which
+also has a role_default attribute set and are being merged.
+
+## Deprecate Access to Automatic Attributes to the Node Object
+
+Rubocop rules will be added to cookstyle to warn when common automatic node attributes are being accessed
+through (e.g.) `node["platform"]` and will have an autofix rule to change them to (e.g.) `ohai("platform")`.
+
+On access the chef-client will issue deprecation warnings to any access to automatic attributes through the
+deep-merged view of the node object.
+
+## Deprecate Access to Normal Attributes to the Node Object
+
+Rubocop rules will be added to cookstyle to warn when common automatic node attributes are being accessed
+through (e.g.) `node["wordpress"]["passwd"]` and will have an autofix rule to change them to (e.g.)
+`store("wordpress", "passwd")`.
+
+On access the chef-client will issue deprecation warnings to any access to normal attributes through the
+deep-merged view of the node object.
+
+## Future:  Deprecate Access to Attributes Through the Node Object
+
+In a future version of Chef (no sooner than Chef 14) access to default and override objects through the node
+object APIs will be deprecated completely in favor of the new APIs.  In the following major release (no
+sonner than Chef 15) the old APIs will be disabled.
```

Comment:
IMO this should be Chef 14 since it will probably be deprecated as of 13.  This gives people 1 year to remove the old API calls.

Counterpoint: this might require a bit of changes to many third-party plugins and tooling